core/embed/rust/src/ui/model_tt/component/button.rs

```diff
@@ -238,3 +252,65 @@ pub struct ButtonStyle {
     pub border_radius: u8,
     pub border_width: i32,
 }
+
+pub struct ButtonArray<T> {
```

Comment:
I have a feeling that the number of these container types will grow, and in some cases a bit of generalisation would be useful. IIRC i've tried to do this with the tuple components, and the `Map`ping wrapper. What do you think?

Reply:
That's a valid concern. I didn't understand what `Map` is for but now I think I do - please take a look at c345d66 which attempts to implement button array in terms of tuples and `Map`.